### ISSUE TITLE
Custom fields added to log records are duplicated when logging a string

### ISSUE THREAD

**Issue by dwarfcrank** [jteppinette/python-logfmter#10]
We use logging filters to add some custom fields to each log message (usually a trace ID) without having to specify it in `extra` every time. This causes the custom fields to be formatted twice if the logged message is a string.

Here is a simple repro case which shows the problem:
```python
import logging
import logging.config

_LOGGING_CONFIG = {
    "version": 1,
    "formatters": {
        "logfmt": {
            "()": "logfmter.Logfmter",
            "keys": ["level", "module", "custom_field1", "custom_field2"],
            "mapping": {
                "level": "levelname",
                "module": "name",
            },
        },
    },
    "handlers": {
        "console": {
            "class": "logging.StreamHandler",
            "formatter": "logfmt",
            "stream": "ext://sys.stdout",
        },
    },
    "loggers": {
        "": {"handlers": ["console"], "level": "INFO"},
    },
    "disable_existing_loggers": False,
}

logging.config.dictConfig(_LOGGING_CONFIG)

_prev_factory = logging.getLogRecordFactory()

def _log_record_factory(*args, **kwargs) -> logging.LogRecord:
    record = _prev_factory(*args, **kwargs)

    record.custom_field1 = "aaa"
    record.custom_field2 = "bbb"

    return record

logging.setLogRecordFactory(_log_record_factory)

if __name__ == "__main__":
    logging.info("this hits the bug")
    logging.info({"msg": "this doesn't"})
```

Output:
```
level=INFO module=root custom_field1=aaa custom_field2=bbb msg="this hits the bug" custom_field1=aaa custom_field2=bbb
level=INFO module=root custom_field1=aaa custom_field2=bbb msg="this doesn't"
```

Logfmt version: 0.0.6 from PyPI

**Comment by jteppinette** [comment]
@dwarfcrank Thank you for the bug report and reproduction. I will take a look at this today or tomorrow.

**Comment by matteo-zanoni** [comment]
I happend to have a similare use case (adding custom fields to the formatter), so I looked into this...

I found a fix for the mentioned bug: it seems that the extra fields are added to the message twice (once in `self.get_extra(record)` and another time in `for key in self.keys`).
I would be happy to open a PR with the fix.

Looking deeper at the issue I could not overlook the fact that @dwarfcrank example looks to me to be a _hacky_ way to solve a problem (that I also faced). I believe a more elegant solution (that would also fix the problem) would be to allow for an extra constructor parameter to `Logfmter.__init__` to allow for _static_ extra fields to be added to each log line
The example in the issue could therefore become something like:

```python
import logging
import logging.config

_LOGGING_CONFIG = {
    "version": 1,
    "formatters": {
        "logfmt": {
            "()": "logfmter.Logfmter",
            "keys": ["level", "module", "custom_field1", "custom_field2"],
            "mapping": {
                "level": "levelname",
                "module": "name",
            },
            "static_mapping": {
                "custom_field1": "aaa",
                "custom_field2": "bbb",
            }
        },
    },
    "handlers": {
        "console": {
            "class": "logging.StreamHandler",
            "formatter": "logfmt",
            "stream": "ext://sys.stdout",
        },
    },
    "loggers": {
        "": {"handlers": ["console"], "level": "INFO"},
    },
    "disable_existing_loggers": False,
}

logging.config.dictConfig(_LOGGING_CONFIG)

if __name__ == "__main__":
    logging.info("log with the extra fields")
```

I have a proof-of-concept implementation and I would also like to open a PR with a proper implementation of this.

Basically looking to help and wondering what solution to implement.

**Comment by matteo-zanoni** [comment]
@jteppinette sounds good to me! I guess `default_extra` is a much better name (let's say mine was only a placeholder).
I will open the new issue!

**Comment by jteppinette** [comment]
@dwarfcrank @matteo-zanoni This has been resolved via `v0.0.7`.
